### ISSUE TITLE
Event publishing inside long procedure gets blocked

### ISSUE THREAD

**Issue by LMiljak** [crossbario/autobahn-python#1017]
My goal is that I want my long procedure to publish an event just before starting.
I have the following Python code (using Twisted, Python 2.7):
```python
import time
from autobahn.twisted import component

class MyClass(object):

    def __init__(self, session):
        self.session = session

    def my_long_procedure(self):
        self.session.publish(u"topic")
        time.sleep(5) # Waiting 5 seconds to simulate long procedure

def on_join(session, _):
    session.register(MyClass(session).my_long_procedure, u"my_procedure")

if __name__ == "__main__":
    comp = component.Component(transports=u"ws://localhost:9000", realm=u"realm")
    comp.on_join(on_join)
    component.run([comp])
```
The issue is that my event gets blocked until the procedure actually finishes (the 5 seconds have passed). An example showing the problem is the following Autobahn|JavaScript code:
```javascript
var autobahn = require('autobahn')

var connection = new autobahn.Connection({
	url: "ws://localhost:9000",
	realm: "rie.manager"
});

connection.onopen = function(session) {
	console.log("Successfully Connected");
	var startTime = new Date().getTime();

	session.subscribe("event", () => {
		var duration = (new Date().getTime() - startTime) / 1000
		console.log("Event got thrown after " + duration + " seconds");
	});

	session.call("my_procedure").then(() => {
		var duration = (new Date().getTime() - startTime) / 1000
		console.log("Long procedure finished after " + duration + " seconds");
	});
};
connection.open();
```
gives the following result:
```
Successfully Connected
Event got thrown after 5.016 seconds
Long procedure finished after 5.016 seconds
```
You would expect the event get thrown right at the start, but it gets blocked until the procedure finishes. I don't have the issue when I register the procedure in a similar way using Autobahn|Javascript
```javascript
connection.onopen = function(session) {
	wait = function(ms) {
		start = new Date().getTime();
		var end = start;
		while (end < start + ms) {
			end = new Date().getTime();
		}
	};

	session.register("my_procedure", () => {
		session.publish("event");
		wait(5000);
	});
};
```
This gives the following output on the client:
```
Successfully Connected
Event got thrown after 0.003 seconds
Long procedure finished after 5.005 seconds
```
To make my procedure even nicer in JavaScript, I could return a Promise, rather than making the procedure blocking:
```javascript
	session.register("my_procedure", () => {
		session.publish("event");
		return new Promise(function(resolve, reject) {
			setTimeout(resolve, 5000);
		});
	});
```

So my question is:
Is there a way in Autobahn|Python to publish an event at the start of a long blocking procedure? Or allow the procedure to return a Deferred similar to how returning a Promise is allowed in Autobahn|JavaScript?

**Comment by LMiljak** [comment]
Found a fix, using twisted's `@inlineCallback` decorator does the trick.
